agent/recorder-agent.c

```diff
@@ -471,9 +471,9 @@ gboolean
   edge_id = get_records (recording_dir, NULL, arg_record_id, arg_from, arg_to,
       builder);
 
-  records = g_variant_new ("a(sxxx)", builder);
+  records = g_variant_builder_end (builder);
 
-  g_variant_builder_unref (builder);
+  records = g_variant_ref_sink (records);
```

Comment:
From the generated code of `hwangsae1_dbus_recorder_interface_complete_lookup_by_record()` it appears it will sink the floating reference itself. Actually, you can pass the GVariant directly into the `complete` function without taking any reference:

```
  hwangsae1_dbus_recorder_interface_complete_lookup_by_record (object,
      invocation, edge_id, g_variant_builder_end (builder));
```

Same the case below.

Reply:
Thanks.